File: neoforge/src/main/java/com/duck/elliemcquinn/storageoptions/neoforge/registration/ModBlocks.java
package com.duck.elliemcquinn.storageoptions.neoforge.registration;

import com.duck.elliemcquinn.storageoptions.neoforge.Main;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockBehaviour;
import net.neoforged.neoforge.registries.DeferredBlock;
import net.neoforged.neoforge.registries.DeferredRegister;

public class ModBlocks {
    private ModBlocks() {
        // Utility class.
    }

    public static final DeferredRegister.Blocks BLOCKS = DeferredRegister.createBlocks(Main.ID);

    public static final DeferredBlock<Block> TALL_BARREL = BLOCKS.register("barrel", () -> new Block(BlockBehaviour.Properties.of()));
}

File: common/src/main/kotlin/com/duck/elliemcquinn/storageoptions/registration/ModBlocks.kt
package com.duck.elliemcquinn.storageoptions.registration

import com.duck.elliemcquinn.storageoptions.block.ClassicChestBlock
import com.duck.elliemcquinn.storageoptions.block.TallBarrelBlock
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.world.level.block.Blocks
import net.minecraft.world.level.block.state.BlockBehaviour

object ModBlocks {
    val BARREL: TallBarrelBlock = Registry.register(
        BuiltInRegistries.BLOCK,
        "ellsso:barrel",
        TallBarrelBlock(BlockBehaviour.Properties.ofFullCopy(Blocks.BARREL))
    )

    val CLASSIC_CHEST: ClassicChestBlock = Registry.register(
        BuiltInRegistries.BLOCK,
        "ellsso:classic_chest",
        ClassicChestBlock(BlockBehaviour.Properties.ofFullCopy(Blocks.OAK_PLANKS))
    )

    fun init() {
        // NO-OP for now.
    }
}